src/transformers/models/xlnet/modeling_xlnet.py

```diff
@@ -1090,13 +1088,6 @@ def forward(
         )
         return_dict = return_dict if return_dict is not None else self.config.use_return_dict
 
-        if "use_cache" in kwargs:
-            warnings.warn(
-                "The `use_cache` argument is deprecated and will be removed in a future version, use `use_mems` instead.",
-                FutureWarning,
-            )
-            use_mems = kwargs["use_cache"]
-
```

Comment:
You should probably keep this for another PR but it's maybe the time to change `use_cache` to `use_mems`.

Reply:
We should keep this here. This has nothing to do really with this PR IMO

Reply:
Ah sorry, ok I understand! You need to remove the `kwargs` to make it work for ONNX. Hmm, I sadly don't think we can do this before Transformers v5. @sgugger what do you think?

Reply:
Nope, no breaking change until v5 indeed.

Reply:
OK, let's keep this PR open until v5 is released. Nevertheless, thank you for working on this @sijunhe - let's revisit it once we're able to safely remove the `kwargs` from the forward pass!

Reply:
Is there a date for the v5 release? :hugs:

Reply:
Ah it's a bit unfortunate that it doesn't work without this breaking change; unfortunately, no release date for v5 yet and it will likely result from a very big change that needs to happen (probably on the tokenizers side). However, as we want to limit as much as possible the breaking changes that we do, we will not push for v5 and instead aim to release it only when we absolutely need to do it.

Is there no workaround for this? I think we could accept a bit of magic monkey patching for this exact use-case if it remains in the realm of the ONNX export and does not affect the rest.

Reply:
It could be awesome yes!

Reply:
Thanks for the feedback @LysandreJik - this is really helpful! 

As you suggest, with monkey patching we could have something like the following inside the `export_pytorch()` (and possibly `export_tensorflow()`) methods:

```
model.forward = forward_without_kwargs(model.forward)
```

where `forward_without_kwargs()` is a function that wraps the original forward pass to strip out the `kwargs`. 

WDYT @sijunhe - would you like to have a go at implementing this?

Reply:
couldn't we otherwise just remove ``kwargs`` and replace it with `use_cache=None` and then raise a warning if `use_cache is not None`?

Reply:
It would expose the deprecated arg though. So if there is a solution without removing the kwargs, I'd prefer it.